File: DSA/Arrays/06 - Challenge: Rearrange Sorted List in Max/solution.py
#!/bin/python3
"""
Module to rearrange a sorted list into max/min form.
"""

def rearrange_max_min(arr):
    """
    Rearranges a sorted list into max/min form:
    [max, min, 2nd max, 2nd min, ...]
    
    Args:
        arr (List[int]): A sorted list of integers.

    Returns:
        List[int]: Rearranged list in max/min format.
    """
    result = []
    left, right = 0, len(arr) - 1

    while left <= right:
        if left != right:
            result.append(arr[right])
            result.append(arr[left])
        else:
            result.append(arr[left])
        left += 1
        right -= 1

    return result

File: DSA/Arrays/06 - Challenge: Rearrange Sorted List in Max/solution.js
#!/usr/bin/env node
/**
 * Module to rearrange a sorted list into max/min form.
 */

/**
 * Rearranges a sorted array into max/min form:
 * [max, min, 2nd max, 2nd min, ...]
 *
 * @param {number[]} arr - A sorted array of integers.
 * @returns {number[]} Rearranged array in max/min format.
 */
function rearrangeMaxMin(arr) {
  const result = [];
  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    if (left !== right) {
      result.push(arr[right]);
      result.push(arr[left]);
    } else {
      result.push(arr[left]);
    }
    left++;
    right--;
  }

  return result;
}
